### PR TITLE
Rename params d3

### PR DESCRIPTION
**Pull request by shawnoden**

### CODE REVIEW

File: data/en/duplicate.json [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 	"params": [
-		{"name":"variable","description":"","required":true,"default":"","type":"any","values":[]}
+		{"name":"object","description":"Name of the object to duplicate.","required":true,"default":"","type":"any","values":[]},
+		{"name":"deepcopy","description":"Lucee Only. \rIf set to `true` (default) the child elements are also cloned. If `false`, child elements retain a reference to their corresponding element in the original object. \rNote: deeply cloned elements that are not native Lucee objects (i.e. Java objects) may change data type when they can be converted to a native CFML object.","required":false,"default":"true","type":"boolean","values":[]}
 	],
 	"engines": {
-		"coldfusion": {"minimum_version":"4.5", "notes":"", "docs":"https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-c-d/duplicate.html"},
+		"coldfusion": {"minimum_version":"4.5", "notes":"CFMX allows this function to be used on XML objects. CF8 allows this to duplicate CFCs. CF2018 introduced named parameters.", "docs":"https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-c-d/duplicate.html"},
```
Comment (KamasamaK) [inline data/en/duplicate.json]: CF2018 introduced named parameters for all functions, so I don't think it's necessary for the notes here.

File: data/en/dotnettocftype.json [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
-	"description":" Explicitly converts a value returned by a .NET method to the corresponding ColdFusion data type.",
+	"description":"Explicitly converts a value returned by a .NET method to the corresponding ColdFusion data type.",
 	"params": [
-		{"name":"variable_name","description":"","required":true,"default":"","type":"Varible","values":[]}
+		{"name":"variableName","description":"Name of the .NET variable to convert.","required":true,"default":"","type":"Varible","values":[]}
 
 	],
 	"engines": {
-		"coldfusion": {"minimum_version":"", "notes":"", "docs":"https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-c-d/dotnettocftype.html"}
+		"coldfusion": {"minimum_version":"8", "notes":"CF2018 changed `variable_name` to `variableName`.", "docs":"https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-c-d/dotnettocftype.html"}
```
Comment (KamasamaK) [inline data/en/dotnettocftype.json]: Named parameters were introduced in CF2018 so I'm not sure what it means to have the name changed.

File: data/en/directoryrename.json [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-		{"name":"newPath","description":"","required":true,"default":"","type":"string","values":[]}
+		{"name":"oldPath","description":"Absolute path of the directory to be renamed. Alternatively, you can specify IP address, for example, `DirectoryRename(\"//12.3.123.123/c_drive/test\");`","required":true,"default":"","type":"string","values":[]},
+		{"name":"newPath","description":"New name for the directory.","required":true,"default":"","type":"string","values":[]},
+		{"name":"createPath","description":"Lucee Only. If set to `false`, expects all parent directories to exist. `true` will generate necessary directories.","required":false,"default":"true","type":"boolean","values":[]}
 	],
 	"engines": {
-		"coldfusion": {"minimum_version":"9", "notes":"", "docs":"https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-c-d/DirectoryRename.html"},
+		"coldfusion": {"minimum_version":"9", "notes":"CF2018 renamed `path` to `oldPath` and `newName` to `newPath`.", "docs":"https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-c-d/DirectoryRename.html"},
```
Comment (KamasamaK) [inline data/en/directoryrename.json]: Named parameters were introduced in CF2018 so I'm not sure what it means to have the name changed.

File: data/en/directorylist.json [hunk 4]
```diff
@@ -71,7 +71,7 @@
 	"engines": {
 		"coldfusion": {
 			"minimum_version": "9",
-			"notes": "Only CF11+ supports the type argument.",
+			"notes": "Only CF11+ supports the type argument. CF2018 added named parameters.",
```
Comment (KamasamaK) [inline data/en/directorylist.json]: CF2018 introduced named parameters for all functions, so I don't think it's necessary for the notes here.

File: data/en/duplicate.json (hunk 1 above)
Comment (shawnoden) [reply data/en/duplicate.json]: Changes made.  CF2018 comment removed.

File: data/en/directoryrename.json (hunk 3 above)
Comment (shawnoden) [reply data/en/directoryrename.json]: Changes made.  CF2018 comment removed.

File: data/en/directorylist.json (hunk 4 above)
Comment (shawnoden) [reply data/en/directorylist.json]: Changes made.  CF2018 comment removed.

File: data/en/dotnettocftype.json (hunk 2 above)
Comment (shawnoden) [reply data/en/dotnettocftype.json]: Changes made.  CF2018 comment removed.

File: data/en/directorylist.json (hunk 4 above)
Comment (KamasamaK) [reply data/en/directorylist.json]: I don't think this was removed.

File: data/en/deserializexml.json [hunk 5]
```diff
@@ -1,19 +1,23 @@
...
+		{"name":"string","description":"A string that needs to be deserialized.","required":true,"default":"","type":"String","values":[]},
+                {"name":"useCustomSerializer","description":"This identifies whether or not to use the custom serializer. The default value is true. The custom serializer will be always used for XML deserialization. If false, the XML/JSON deserialization will be done using the default ColdFusion behavior. If any other type is passed with `useCustomSerializer` as false, then `TypeNotSupportedException` will be thrown.","required":true,"default":"true","type":"boolean","values":[]}
 	],
 	"engines": {
-		"coldfusion": {"minimum_version":"11", "notes":"", "docs":"https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-c-d/deserializexml.html"}
+		"coldfusion": {"minimum_version":"11", "notes":"CF2018 added named parameters.", "docs":"https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-c-d/deserializexml.html"}
```
Comment (KamasamaK) [inline data/en/deserializexml.json]: No need for the notes here

File: data/en/directorylist.json (hunk 4 above)
Comment (shawnoden) [reply data/en/directorylist.json]: Updated. Check now. Not sure what happened there.

File: data/en/deserializexml.json (hunk 5 above)
Comment (shawnoden) [reply data/en/deserializexml.json]: Made recommended changes.

**Comment by shawnoden** [comment]
Also this one. Was there something else in this PR that needed to be completed?

**Comment by pfreitag** [comment]
Looked ok to me, @KamasamaK were all your concerns addressed?

**Comment by KamasamaK** [comment]
Yes, they were addressed.

**Comment by pfreitag** [comment]
Thanks @shawnoden for your contribution, and @KamasamaK for your help reviewing this!

**Pull request merged**